src/modules/template-details/TemplateDetails.tsx

```diff
@@ -32,6 +34,8 @@ const TemplateDetails = ({ match }: TemplateDetails.Props) => {
 
   const { template, loading, getTemplateDetails } = useTemplateDetailsProvider();
 
+  const { user, authorized } = useAuthProvider();
```

Comment:
It works. But imagine you need to do same situation in other component / place. This solution is not reusable and adds additional logic to TempaltesDetails. My recommendation is - create component same as one of 4 already created guards. Use auth provider inside and add needd checks. You can easy reuse that in whole app. This guard should be moved to Guards.tsx file as additional function inside this file and give it name OnlyAuthor. Check Guard.tsx file for ref